packages/vite/src/node/config.ts

```diff
@@ -577,8 +606,14 @@ export async function resolveConfig(
 
   const BASE_URL = resolvedBase
 
-  const resolved: ResolvedConfig = {
-    ...config,
+  // resolve worker
+  const resolvedWorkerOptions: ResolveWorkerOptions = {
+    format: workerConfig.worker?.format || 'iife',
+    plugins: [],
+    rollupOptions: workerConfig.worker?.rollupOptions || {}
+  }
```

Comment:
**NOTE**: the raw `userPlugin` can not modify the `config.worker`. 

`user config config` and `worker resolved config` use the same `resolvedWorkerOptions` what config by `vite.config.js` and modify by worker plugin.